sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/datasources/subscriptions/SubscriptionsDbConnector.groovy

```diff
@@ -116,4 +120,32 @@ class SubscriptionsDbConnector implements SubscriptionDataSource {
           }
           return personId
     }
+
+    private boolean projectAlreadySubscribed(int subscriberId, String projectCode) {
+        String query = "SELECT id FROM subscription WHERE project_code = ? AND subscriber_id = ? "
+        Connection connection = connectionProvider.connect()
+        boolean isAlreadySubscribed = false
+
+        try{
+            connection.withCloseable {
+                def statement = connection.prepareStatement(query)
+                statement.setString(1, projectCode)
+                statement.setString(2, subscriberId.toString())
+                ResultSet resultSet = statement.executeQuery()
+                int numberOfRows = 0
+                while(resultSet.next()) {
+                    numberOfRows++
+                }
+                if (numberOfRows > 0) {
+                    isAlreadySubscribed = true
+                }
+            }
+        }catch(Exception e){
+            log.error(e.message)
+            log.error(e.stackTrace.join("\n"))
+            connection.rollback()
+            throw new DataSourceException("Could not check if the subscriber is subscribed to project ${projectCode}")
+        }
+        return isAlreadySubscribed
+    }
```

Comment:
I think you can neglect the try/catch here, it will be called in the calling method

```suggestion
    private boolean projectAlreadySubscribed(int subscriberId, String projectCode) {
        String query = "SELECT id FROM subscription WHERE project_code = ? AND subscriber_id = ? "
        Connection connection = connectionProvider.connect()
        boolean isAlreadySubscribed = false
            connection.withCloseable {
                def statement = connection.prepareStatement(query)
                statement.setString(1, projectCode)
                statement.setString(2, subscriberId.toString())
                ResultSet resultSet = statement.executeQuery()
                int numberOfRows = 0
                while(resultSet.next()) {
                    numberOfRows++
                }
                if (numberOfRows > 0) {
                    isAlreadySubscribed = true
                }
            }
        return isAlreadySubscribed
    }
```